packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -1164,6 +1164,32 @@ export class ChannelCollection implements IFluidDataStoreChannel, IDisposable {
 		}
 	}
 
+	/**
+	 * Notifies all data store contexts about the current staging mode state.
+	 *
+	 * @param staging - A boolean indicating whether the container is in staging mode.
+	 */
+	public notifyStagingMode(staging: boolean): void {
+		for (const [fluidDataStoreId, context] of this.contexts) {
+			try {
+				context.notifyStagingMode(staging);
+			} catch (error) {
```

Comment:
I like the idea of try/catch since this is ultimately invoking customer code which might have some random error, but would maybe make more sense to implement the check closer to the point where it's going to actually invoke customer code?  Thinking in FluidDataStoreContext for example?